sql/hive/src/test/scala/org/apache/spark/sql/hive/ParquetEncryptionSuite.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive
+
+import java.io.File
+import java.io.RandomAccessFile
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+
+/**
+ * A test suite that tests parquet modular encryption usage.
+ */
+class ParquetEncryptionSuite extends QueryTest with TestHiveSingleton {
+  import spark.implicits._
+
+  private val encoder = Base64.getEncoder
+  private val footerKey =
+    encoder.encodeToString("0123456789012345".getBytes(StandardCharsets.UTF_8))
+  private val key1 = encoder.encodeToString("1234567890123450".getBytes(StandardCharsets.UTF_8))
+  private val key2 = encoder.encodeToString("1234567890123451".getBytes(StandardCharsets.UTF_8))
+
+  test("SPARK-34990: Write and read an encrypted parquet") {
+    withTempDir { dir =>
+      withSQLConf(
+        "parquet.crypto.factory.class" ->
+          "org.apache.parquet.crypto.keytools.PropertiesDrivenCryptoFactory",
+        "parquet.encryption.kms.client.class" ->
+          "org.apache.parquet.crypto.keytools.mocks.InMemoryKMS",
+        "parquet.encryption.key.list" ->
+          s"footerKey: ${footerKey}, key1: ${key1}, key2: ${key2}") {
+
+        val inputDF = Seq((1, 22, 333)).toDF("a", "b", "c")
+        val parquetDir = new File(dir, "parquet").getCanonicalPath
+        inputDF.write
+          .option("parquet.encryption.column.keys", "key1: a, b; key2: c")
+          .option("parquet.encryption.footer.key", "footerKey")
+          .parquet(parquetDir)
+
+        verifyParquetEncrypted(parquetDir)
+
+        val parquetDF = spark.read.parquet(parquetDir)
+        assert(parquetDF.inputFiles.nonEmpty)
+        val readDataset = parquetDF.select("a", "b", "c")
+        checkAnswer(readDataset, inputDF)
+      }
+    }
+  }
+
+  private def verifyParquetEncrypted(parquetDir: String) = {
+    val parquetPartitionFiles = getListOfParquetFiles(new File(parquetDir))
+    assert(parquetPartitionFiles.size >= 1)
```

Comment:
This is always 1 as your test writing the data into a brand new temporary directory:
https://github.com/apache/spark/blob/aa507a47fbbe6e034726448f676df841754c6734/sql/hive/src/test/scala/org/apache/spark/sql/hive/ParquetEncryptionSuite.scala#L41

I have checked out your PR and run test just to be sure.

That temp directory contained only one **parquet** file:
- /Users/attilazsoltpiros/git/attilapiros/spark/target/tmp/spark-9998c611-24e0-4147-bb60-625f6bfa11ea/parquet/part-00000-dffeefb2-e638-4e85-a42c-3763ae8919e3-c000.snappy.parquet

The others are filtered out by `getListOfParquetFiles()` as their suffixes are different:
- /Users/attilazsoltpiros/git/attilapiros/spark/target/tmp/spark-9998c611-24e0-4147-bb60-625f6bfa11ea/parquet/._SUCCESS.crc
- /Users/attilazsoltpiros/git/attilapiros/spark/target/tmp/spark-9998c611-24e0-4147-bb60-625f6bfa11ea/parquet/.part-00000-dffeefb2-e638-4e85-a42c-3763ae8919e3-c000.snappy.parquet.crc
- /Users/attilazsoltpiros/git/attilapiros/spark/target/tmp/spark-9998c611-24e0-4147-bb60-625f6bfa11ea/parquet/_SUCCESS)

Reply:
And of course as a consequence in line 70 you do not need the `maxBy`.

Reply:
You're right, @attilapiros , but I think that this method would be useful for other ParquetEncryptionSuite tests that could be added in the future if it were a bit more generic. How about if it loops over all the parquet part files in the folder and checks that all of them have the correct magic string?

Reply:
I see in this case let's change this `maxBy` to a `foreach { parquetFile =>`.